tox.ini

```diff
@@ -2,17 +2,27 @@
 envlist =
     {py36,py37,py38,py39}-django{22,31,32},
     {py38,py39}-django-latest,
+    lint
 
 [testenv]
 deps =
     django22: django>=2.2,<2.3
     django31: django>=3.1,<3.2
     django32: django>=3.2,<3.3
     django-latest: https://github.com/django/django/archive/main.tar.gz
-    -rrequirements.txt
+    -rrequirements/testing.txt
 allowlist_externals = make
-commands = make test
+commands = python -W error::DeprecationWarning -W error::PendingDeprecationWarning -m pytest {posargs} --cov=crispy_forms
 ignore_outcome =
     django-latest: True
 setenv =
     PYTHONDONTWRITEBYTECODE=1
+
+[testenv:lint]
```

Comment:
don't add a `lint` environment - just run `pre-commit` in a github action?

See e.g. https://github.com/django/daphne/pull/365

Reply:
Or maybe have the lint environ use pre-commit... 🤔

(My muscle memory is objecting to not having it already. :D)

Reply:
Well up to you

Reply:
Now I'm really unsure! I also like the idea of pre-commit.ci 🤷‍♂️ 

I guess let's have this as it is, this is significantly better than what we have at the moment.